### PR TITLE
Adding support for Barduino Board

### PR DESCRIPTION
**Pull request by JosepMartiElias**
Adding Barduino PID's.

Barduino is a development board created by Fablab Barcelona as a tool for it's educational programs.

ESP32-S3 (ESP32-S3-WROOM-1)

It needs a custom PID to have the opportunity to show it officialy in the CircuitPython website and to create an Arduino core for it.

It's done under Fablab Barcelona, part of IAAC.

This is (will be because it's a WIP) the documentation for the board https://fablabbcn-projects.gitlab.io/electronics/barduino-docs/
and this is Fablab Barcelona https://fablabbcn.org/
and this IAAC https://iaac.net/

Thank you!

### CODE REVIEW

File: ports/espressif/boards/barduino/pins.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// SPDX-FileCopyrightText: Copyright (c) 2020 Scott Shawcroft for Adafruit Industries
+//
+// SPDX-License-Identifier: MIT
+
+#include "shared-bindings/board/__init__.h"
+
+static const mp_rom_map_elem_t board_module_globals_table[] = {
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
+
+    { MP_ROM_QSTR(MP_QSTR_IO0), MP_ROM_PTR(&pin_GPIO0) },
```
Comment (dhalbert) [inline ports/espressif/boards/barduino/pins.c:12]: Could you group aliases for a pin together, and put the preferred name first, with a blank line between groups. For example:
```suggestion
    { MP_ROM_QSTR(MP_QSTR_BUTTON), MP_ROM_PTR(&pin_GPIO0) },
    { MP_ROM_QSTR(MP_QSTR_IO0), MP_ROM_PTR(&pin_GPIO0) },
    
```
and remove `BUTTON` from the end of the list. This makes it easier to check to and to find the pin names.

File: ports/espressif/boards/barduino/pins.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_GPIO43) },
+    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_GPIO44) },
+
+    { MP_ROM_QSTR(MP_QSTR_SDA), MP_ROM_PTR(&pin_GPIO8) },
+    { MP_ROM_QSTR(MP_QSTR_SCL), MP_ROM_PTR(&pin_GPIO9) },
+
+    { MP_ROM_QSTR(MP_QSTR_SCK), MP_ROM_PTR(&pin_GPIO12) },
+    { MP_ROM_QSTR(MP_QSTR_MOSI), MP_ROM_PTR(&pin_GPIO11) },
+    { MP_ROM_QSTR(MP_QSTR_MISO), MP_ROM_PTR(&pin_GPIO13) },
```
Comment (dhalbert) [inline ports/espressif/boards/barduino/pins.c:62]: You gave default names for I2C and SPI for these pins, but didn't define `board.I2C()` and `board.SPI()`, only `board.UART()`. Normally if you have default pins for these devices, they'd have a `board` attribute.

File: ports/espressif/boards/barduino/pins.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    { MP_ROM_QSTR(MP_QSTR_IO43), MP_ROM_PTR(&pin_GPIO43) },
+    { MP_ROM_QSTR(MP_QSTR_IO44), MP_ROM_PTR(&pin_GPIO44) },
+    { MP_ROM_QSTR(MP_QSTR_IO45), MP_ROM_PTR(&pin_GPIO45) },
+    { MP_ROM_QSTR(MP_QSTR_IO46), MP_ROM_PTR(&pin_GPIO46) },
+    { MP_ROM_QSTR(MP_QSTR_IO47), MP_ROM_PTR(&pin_GPIO47) },
+    { MP_ROM_QSTR(MP_QSTR_IO48), MP_ROM_PTR(&pin_GPIO48) },
+    { MP_ROM_QSTR(MP_QSTR_LED), MP_ROM_PTR(&pin_GPIO48) },
+    { MP_ROM_QSTR(MP_QSTR_NEOPIXEL), MP_ROM_PTR(&pin_GPIO38) },
+    { MP_ROM_QSTR(MP_QSTR_BUZZER), MP_ROM_PTR(&pin_GPIO48) },
+    { MP_ROM_QSTR(MP_QSTR_PT), MP_ROM_PTR(&pin_GPIO3) },
```
Comment (dhalbert) [inline ports/espressif/boards/barduino/pins.c:51]: What is `PT`? Thanks.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Hi - just some `pins.c` questions and requests.

File: ports/espressif/boards/barduino/pins.c (hunk 2 above)
Comment (JosepMartiElias) [reply ports/espressif/boards/barduino/pins.c]: Done, I've also added I2C and SPI definitions, thank you!

File: ports/espressif/boards/barduino/pins.c (hunk 3 above)
Comment (JosepMartiElias) [reply ports/espressif/boards/barduino/pins.c]: It's a Phototransistor, I've added a comment also.

File: ports/espressif/boards/barduino/pins.c [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+static const mp_rom_map_elem_t board_module_globals_table[] = {
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
+
+    { MP_ROM_QSTR(MP_QSTR_BUTTON), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_IO0), MP_ROM_PTR(&pin_GPIO0) },
+
+    { MP_ROM_QSTR(MP_QSTR_IO1), MP_ROM_PTR(&pin_GPIO1) },
+    { MP_ROM_QSTR(MP_QSTR_IO2), MP_ROM_PTR(&pin_GPIO2) },
+
+    { MP_ROM_QSTR(MP_QSTR_PT), MP_ROM_PTR(&pin_GPIO3) }, // PT = phototransistor
```
Comment (dhalbert) [inline ports/espressif/boards/barduino/pins.c:18]: The standard name on other boards for this is `LIGHT`, short for "light sensor"
```suggestion
    { MP_ROM_QSTR(MP_QSTR_LIGHT), MP_ROM_PTR(&pin_GPIO3) }, // PT = phototransistor
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
One more suggestion: `PT` -> `LIGHT`

**Review by dhalbert — APPROVED** [review]
Thanks for your patience!

**Pull request merged**